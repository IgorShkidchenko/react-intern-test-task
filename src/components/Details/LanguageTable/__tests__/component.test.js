import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import LanguageRowComponent from "../../LanguageRow/component";
import LanguageTableComponent from "../component";

configure({ adapter: new Adapter() });

describe("<LanguageTableComponent />", () => {
  it("renders correctly", () => {
    const rows = [
      <LanguageRowComponent percentage="1" languageName="name" key="name" />,
      <LanguageRowComponent percentage="2" languageName="name2" key="name2" />,
    ];
    const wrapper = shallow(<LanguageTableComponent rows={rows} />);

    expect(wrapper).toMatchSnapshot();
  });
});
