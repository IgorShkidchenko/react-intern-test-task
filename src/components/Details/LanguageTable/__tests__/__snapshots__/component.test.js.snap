// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageTableComponent /> renders correctly 1`] = `
<table>
  <thead>
    <tr>
      <td>
        Languages:
      </td>
    </tr>
  </thead>
  <tbody>
    <LanguageRowComponent
      key="name"
      languageName="name"
      percentage="1"
    />
    <LanguageRowComponent
      key="name2"
      languageName="name2"
      percentage="2"
    />
  </tbody>
</table>
`;
